pages/api/applications/[applicationId]/stages/index.ts

```diff
@@ -1,5 +1,133 @@
 import type { NextApiRequest, NextApiResponse } from 'next';
+import { Application, ApplicationStage, ApplicationStageType, PrismaClient } from '@prisma/client';
+import { ApiResponse, StatusMessageType } from '../../../../../types/apiResponse';
+import { createJsonResponse, HttpMethod, HttpStatus, rejectHttpMethod } from '../../../../../utils/http/httpHelpers';
+import { withPrismaErrorHandling } from '../../../../../utils/prisma/prismaHelpers';
+import { ApplicationStageData, ApplicationStagePostData } from '../../../../../types/applicationStage';
+import { isValidDate } from '../../../../../utils/date/validations';
+import { withAuthUser } from '../../../../../utils/auth/jwtHelpers';
+import { Nullable } from '../../../../../types/utils';
 
-export default function handler(req: NextApiRequest, res: NextApiResponse) {
-  res.status(200).end(`Request sent to ${req.url}\n`);
+const { ERROR, SUCCESS } = StatusMessageType;
+const prisma = new PrismaClient();
+
+type ApplicationDataWithStages = Application & { applicationStages: ApplicationStage[] };
+
+enum MessageType {
+  INVALID_TYPE,
+  INVALID_DATE,
+  INVALID_APPLICATION_ID,
+  APPLICATION_NOT_FOUND,
+  CREATED,
+}
+
+const messages = Object.freeze({
+  [MessageType.INVALID_TYPE]: {
+    type: ERROR,
+    message:
+      'Application Stage type is invalid. Type must be one of: ' + Object.values(ApplicationStageType).join(', ') + '.',
+  },
+  [MessageType.INVALID_DATE]: {
+    type: ERROR,
+    message: 'Application Stage date is invalid.',
+  },
+  [MessageType.INVALID_APPLICATION_ID]: {
+    type: ERROR,
+    message: 'Query parameter applicationId (.../applications/:applicationId/...) is invalid.',
+  },
+  [MessageType.APPLICATION_NOT_FOUND]: {
+    type: ERROR,
+    message: 'User does not have any application with the given ID.',
+  },
+  [MessageType.CREATED]: { type: SUCCESS, message: `Application stage has been created.` },
+});
+
+async function handler(
+  uid: string,
+  req: NextApiRequest,
+  res: NextApiResponse<ApiResponse<ApplicationStageData | ApplicationDataWithStages>>,
+) {
+  const errorMessageIfInvalid = validatePathParameters(req);
+  if (errorMessageIfInvalid !== null) {
+    return res.status(HttpStatus.BAD_REQUEST).json(createJsonResponse({}, messages[errorMessageIfInvalid]));
+  }
+
+  const application = await getApplication(uid, Number(req.query.applicationId));
+  if (application === null) {
+    return res.status(HttpStatus.NOT_FOUND).json(createJsonResponse({}, messages[MessageType.APPLICATION_NOT_FOUND]));
+  }
+
+  switch (req.method) {
+    case HttpMethod.POST:
+      return postHandler(application, req, res);
+    case HttpMethod.GET:
+      return getHandler(application, res);
```

Comment:
```suggestion
      return handleGet(application, res);
```

nit: for standardisation. also can move this above?